### PR TITLE
Add libe57format/2.2.0

### PR DESCRIPTION
**Pull request by vcampmany**
Specify library name and version:  **libe57format/2.2.0**

Library for reading & writing the E57 file format: https://github.com/asmaloney/libE57Format

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libe57format/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```
Comment (uilianries) [inline recipes/libe57format/all/conanfile.py:50]:
```suggestion
```

Automatically solved by CMake helper when the option shared is available.

File: recipes/libe57format/all/patches/fix-cmake-package-files.patch [hunk 2]
```diff
@@ -0,0 +1,19 @@
+--- a/CMakeLists.txt
```
Comment (uilianries) [inline recipes/libe57format/all/patches/fix-cmake-package-files.patch:1]: Remove this patch. Remove the installed cmake files directly.

File: recipes/libe57format/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+class LibE57FormatConan(ConanFile):
+    name = "libe57format"
+    license = "BSL-1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/asmaloney/libE57Format"
+    description = "Library for reading & writing the E57 file format"
+    topics = ("e57", "io", "point-cloud")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
```
Comment (uilianries) [inline recipes/libe57format/all/conanfile.py:14]: fPIC is not honored, need a patch to remove it from upstream's cmakefile:
https://github.com/asmaloney/libE57Format/blob/v2.2.0/CMakeLists.txt#L107

File: recipes/libe57format/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["E57Format"]
```
Comment (uilianries) [inline recipes/libe57format/all/conanfile.py:69]:
```suggestion
        self.cpp_info.libs = ["E57Format-d" if self.settings.build_type == "Debug" else "E57Format"]
```

https://github.com/asmaloney/libE57Format/blob/v2.2.0/CMakeLists.txt#L106

File: recipes/libe57format/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        self.cpp_info.names["cmake_find_package"] = "E57Format"
+        self.cpp_info.names["cmake_find_package_multi"] = "E57Format"
+        self.cpp_info.names["pkg_config"] = "E57Format"
```
Comment (uilianries) [inline recipes/libe57format/all/conanfile.py:72]:
```suggestion
        self.cpp_info.filenames["cmake_find_package"] = "e57format"
        self.cpp_info.filenames["cmake_find_package_multi"] = "e57format"
        self.cpp_info.names["pkg_config"] = "e57format"
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.append("pthread")
```

File: recipes/libe57format/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibE57FormatConan(ConanFile):
```
Comment (madebr) [inline recipes/libe57format/all/conanfile.py:5]:
```suggestion
from conans.errors import ConanInvalidConfiguration

required_conan_version = ">=1.33.0"


class LibE57FormatConan(ConanFile):
```

File: recipes/libe57format/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        if self.settings.os == "Windows":
+            self._cmake.definitions["USING_STATIC_XERCES"] = True
```
Comment (madebr) [inline recipes/libe57format/all/conanfile.py:52]: Is there a reason this is only needed on Windows?
You can use `not self.options["xerces-c"].shared`

File: recipes/libe57format/all/conanfile.py (hunk 5 above)
Comment (madebr) [reply recipes/libe57format/all/conanfile.py:72]: I don't find any mention of a `.pc` file in the [cmake script](https://github.com/asmaloney/libE57Format/blob/b5b76616836754012cc8e7809bd19cee00539d3c/CMakeLists.txt#L132-L154).
So I'ld remove `self.cpp_info.names["pkg_config"] = "e57format"`

File: recipes/libe57format/all/test_package/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+
+find_package(E57Format)
```
Comment (madebr) [inline recipes/libe57format/all/test_package/CMakeLists.txt:9]:
```suggestion
find_package(E57Format REQUIRED)
```

File: recipes/libe57format/all/test_package/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (madebr) [inline recipes/libe57format/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

File: recipes/libe57format/all/conanfile.py (hunk 5 above)
Comment (uilianries) [reply recipes/libe57format/all/conanfile.py:72]: Agreed

File: recipes/libe57format/all/patches/fix-cmake-package-files.patch (hunk 2 above)
Comment (vcampmany) [reply recipes/libe57format/all/patches/fix-cmake-package-files.patch]: Fixed

File: recipes/libe57format/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
```
[the remark this answers is not in the document]
Comment (vcampmany) [reply recipes/libe57format/all/conanfile.py:69]: Fixed.

File: recipes/libe57format/all/conanfile.py (hunk 3 above)
Comment (vcampmany) [reply recipes/libe57format/all/conanfile.py:17]: Thanks, wrote a patch for it.

File: recipes/libe57format/all/conanfile.py (hunk 7 above)
Comment (vcampmany) [reply recipes/libe57format/all/conanfile.py]: You are right, changed it to your suggestion

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**